File: binary_search.js


function binarySearch(items, value) {
    // let itemCount = items.length;
    let start = 0
    end = items.length - 1
    let mid = Math.floor((end+start) / 2)


    while (value !== items[mid] && start < end) {
        console.log(mid)
        if (value > items[mid]) {
            start = mid + 1;
        } else {
            end = mid - 1;
        }
        mid = Math.floor((end+start) / 2)
    }

    if (items[mid] == value) {
        return items[mid];
    }
    return -1;
}

console.log(binarySearch([1, 2, 3, 4, 5,4], 10))

File: binary_search.dart
void main() {
  final arr = [1, 2, 3, 4, 5, 6, 7, 8, 9];

  int start = 0;
  int end = arr.length;
  print(binarySearch(arr, start, end-1, 11));
}

bool binarySearch(arr, start, end, target) {
  if (start > end ) return false;
  int midIndex = arr.length ~/ 2;
  if (arr[midIndex] == target) return true;
  if (arr[midIndex] < target)
    return binarySearch(arr, midIndex + 1, end, target);
  if (arr[midIndex] > target) return binarySearch(arr, 0, midIndex - 1, target);
  return false;
}


